### ISSUE TITLE
Unable to obtain historical stake

### ISSUE THREAD

**Issue by mcdee** [casper-network/casper-node#1270]
I'm attempting to obtain historical (balance+stake) information from the JSON-RPC interface.  I can obtain historical balance (which is currently always 0 as all balance is staked) and current stake (from the state_get_auction_info method) but not historical stake.

Is historical stake available somewhere?  Or some other method to calculate an account's (balance+stake), for example obtaining per-era reward data?

**Comment by piotr-dziubecki** [comment]
Please check the following RPC endpoints to see if they provide you the data you want. 
```
chain_get_era_info_by_switch_block
state_get_auction_info
```

They're also available through the client: 
```
get-era-info-by-switch-block
get-auction-info
```

**Comment by mcdee** [comment]
@piotr-dziubecki the former is good for some of the data.  But the latter does not take a block identifier, and so cannot obtain historical information.  If `state_get_auction_info` took a block identifier (defaulting to the current block to retain existing functionality) then this would solve the issue.

I have a locally tweaked copy of the node that takes a block identifier for `state_get_auction_info` and am happy to share the code as a PR, although I'm sure that Fraser will be able to add this functionality more cleanly than I.

**Comment by Fraser999** [comment]
@mcdee Thanks for the report.  We're planning to improve the `state_get_auction_info` RPC as per your suggestion.  Note that while we'll aim to get the code written and merged next week, it'll be a month or more before it's available on mainnet.

**Comment by Fraser999** [comment]
@mcdee This issue has been addressed in #1459 and will be included in the 1.3.0 release.  Thanks again for the report.
